svm/api.go

```diff
@@ -134,3 +138,67 @@ func (svm *SVM) ValidateAndAddTxToPool(tx *types.Transaction) error {
 	}
 	return nil
 }
+
+// HandleGossipTransaction wraps around HandleTransaction,
+// which handles data received on the transactions gossip channel.
+func (svm *SVM) HandleGossipTransaction(ctx context.Context, _ p2p.Peer, msg []byte) pubsub.ValidationResult {
+	tx, err := types.BytesToTransaction(msg)
+	if err != nil {
+		svm.state.With().Error("SVM couldn't parse incoming transaction", log.Err(err))
+		return pubsub.ValidationIgnore
+	}
+
+	if err = svm.handleTransaction(tx, true); err != nil {
+		svm.state.With().Error("handle transaction", log.Err(err))
+		return pubsub.ValidationIgnore
+	}
+
+	return pubsub.ValidationAccept
+}
+
+// HandleTxSyncData handles data received on transaction sync.
+func (svm *SVM) HandleTxSyncData(data []byte) error {
+	var tx mesh.DbTransaction
+	err := types.BytesToInterface(data, &tx)
+	if err != nil {
+		svm.state.With().Error("SVM couldn't parse incoming transaction", log.Err(err))
+		return fmt.Errorf("parse: %w", err)
+	}
+	if err = tx.CalcAndSetOrigin(); err != nil {
+		return fmt.Errorf("calculate and set origin: %w", err)
+	}
+	svm.state.AddTxToPool(tx.Transaction)
+	return nil
+}
+
+// handleTransaction handles data received on transaction gossip channel.
+func (svm *SVM) handleTransaction(tx *types.Transaction, validateOrigin bool) error {
```

Comment:
Because this is now only ever called from one place (`HandleGossipTransaction`) we can just inline this code there - I don't see the advantage of extracting this logic anymore.

There's also no longer a need for the `validateOrigin` argument which is always set to `true`.